index.js

```diff
@@ -59,3 +77,5 @@ module.exports = function requireAll(options) {
 function identity(val) {
   return val;
 }
+
+function isObject(arg) { return Object.prototype.toString.call(arg).indexOf('Object') !== -1; }
```

Comment:
just put it in to make the actual check cleaner and yes there is a stackoverflow post on this.  Don't have the link at the moment but it does cover extra unusual cases